frontend/src/components/KnowledgeInformation.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <ElDialog :model-value="props.visible" :title="dialogTitle" width="500" @close="hideDialog">
+    <ElInput v-model="localKnowledge.name" placeholder="Enter Name" autocomplete="on" />
+    <div class="Slider">
+      <span class="KnowledgeWeight"> Weight </span>
+      <ElSlider v-model="localKnowledge.weight" :max="5" :step="1" show-stops />
+    </div>
+    <template #footer>
+      <div class="dialog-footer">
+        <ElButton @click="cancelEdit">Cancel</ElButton>
+        <ElButton type="primary" @click="saveForm">{{ dialogButtonLabel }}</ElButton>
+      </div>
+    </template>
+  </ElDialog>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+import { ElButton, ElDialog, ElInput, ElSlider } from 'element-plus'
+import { useKnowledgeStore } from '@/stores/knowledges'
+import { HttpServerError } from '@/services/http'
+import { SuccessMessage, ErrorMessage } from '@/services/messages'
+import type { Knowledge } from '@/domain/Knowledge'
+
+const knowledgeStore = useKnowledgeStore()
+const props = defineProps<{
+  visible: boolean
+  knowledge: Knowledge | undefined
+  roleId: number
+}>()
+const emits = defineEmits(['update:visible', 'confirm-edit', 'confirm-create'])
+const localKnowledge = ref<Knowledge>({ name: '', weight: 0, roleId: props.roleId })
+const isEditMode = ref(false)
+
+watch(
+  () => props.visible,
+  (newVal) => {
+    if (newVal) {
+      if (props.knowledge) {
```

Comment:
You have the same effect doing like this

```suggestion
      localKnowledge.value = !!props.knowledge ?? { ...props.knowledge } : { name: '', weight: 0, roleId: props.roleId }
isEditMode.value = !!props.knowledge
```